lib/NodeGraphEditor.tsx

```diff
@@ -23,24 +24,50 @@ import {
   useMemo,
   JSX,
   CSSProperties,
+  useState,
+  useEffect,
 } from 'react'
 import { defaultEdgeTypes } from './edge-types'
 import { IGraphConfig } from './config'
 import { useSocketConnect } from './hooks/connect'
 import { useHotkeys } from 'react-hotkeys-hook'
 import { ClipboardItem } from './clipboard'
-import { LayoutEngine, useLayoutEngine } from './layout/layout'
+import { LayoutEngine, getLayoutFunction, useLayoutEngine } from './layout/layout'
+
+const options = {
+  includeHiddenNodes: false,
+};
+ 
+export default function useDefaultLayout(layoutFn:((nodes: Node[], edges:Edge[]) => Node[])| undefined) {
+  const { getNodes, getEdges } = useReactFlow();
+  const nodesInitialized = useNodesInitialized(options);
+  const [layoutedNodes, setLayoutedNodes] = useState(getNodes());
+  
+  useEffect(() => {
+    if (nodesInitialized && typeof layoutFn == "function") {
+      const found = !!getNodes().find((node) => node.position == undefined);
+      if (found) {
+        setLayoutedNodes(layoutFn(getNodes(), getEdges()));
+      } else {
+        setLayoutedNodes(getNodes());
+      }
+    }
+  }, [nodesInitialized]);
+ 
+  return layoutedNodes;
+}
 
 type NodeGraphEditorProps = Omit<FlowProps, 'edges' | 'nodes'> & {
-  onSave?: (data: any) => void
+  onSave?: (data: any) => void,
+  defaultLayout?: LayoutEngine,
 }
 
 export const NodeGraphEditor = forwardRef<
   NodeGraphHandle,
   NodeGraphEditorProps
 >(
   (
-    { defaultNodes, defaultEdges, ...props }: NodeGraphEditorProps,
+    { defaultNodes, defaultEdges, defaultLayout, ...props }: NodeGraphEditorProps,
```

Comment:
Not required since we're passing `{...props}` to the `Flow` component so it should already be included.
```suggestion
    { defaultNodes, defaultEdges, ...props }: NodeGraphEditorProps,
```